docs/guides/tms.rst

```diff
@@ -0,0 +1,252 @@
+                       31 : "#BFB8B1FF",     # Bare Rock/Sand/Clay
+                       32 : "#969798FF",     # Quarries/Strip Mines/Gravel Pits
+                       33 : "#382959FF",     # Transitional
+                       41 : "#579D57FF",     # Deciduous Forest
+                       42 : "#2A6B3DFF",     # Evergreen Forest
+                       43 : "#A6BF7BFF",     # Mixed Forest
+                       51 : "#BAA65CFF",     # Shrubland
+                       61 : "#45511FFF",     # Orchards/Vineyards/Other
+                       71 : "#D0CFAAFF",     # Grasslands/Herbaceous
+                       81 : "#CCC82FFF",     # Pasture/Hay
+                       82 : "#9D5D1DFF",     # Row Crops
+                       83 : "#CD9747FF",     # Small Grains
+                       84 : "#A7AB9FFF",     # Fallow
+                       85 : "#E68A2AFF",     # Urban/Recreational Grasses
+                       91 : "#B6D8F5FF",     # Woody Wetlands
+                       92 : "#B6D8F5FF" }    # Emergent Herbaceous Wetlands
+
+   def rgba_functions(color_map):
+      m = {}
+      for key in color_map:
+         m[key] = hex_to_rgb(color_map[key])
+
+
+      def r(v):
+         if v in m:
+            return m[v][0]
+         else:
+            return 0
+
+      def g(v):
+         if v in m:
+            return m[v][1]
+         else:
+            return 0
+
+      def b(v):
+         if v in m:
+            return m[v][2]
+         else:
+            return 0
+
+      def a(v):
+         if v in m:
+            return m[v][3]
+         else:
+            return 0x00
+
+      return (np.vectorize(r), np.vectorize(g), np.vectorize(b), np.vectorize(a))
+
+   def render_nlcd(tile):
+      '''
+      Assumes that the tile is a multiband tile with a single band.
+      (meaning shape = (1, cols, rows))
+      '''
+      arr = tile.cells[0]
+      (r, g, b, a) = rgba_functions(nlcd_color_map)
+
+      rgba = np.dstack([r(arr), g(arr), b(arr), a(arr)]).astype('uint8')
+
+      img = Image.fromarray(rgba, mode='RGBA')
+
+      return img
+
+   tms = gps.TMS.build(('s3://azavea-datahub/catalog', 'nlcd-tms-epsg3857'), display=render_nlcd)
+
+You will likely observe noticeably slower performance compared to the earlier
+example.  This is because the contents of each tile must be transferred from
+the JVM to the Python environment prior to rendering.  If performance is
+important to you, and a color mapping solution is available, please use that
+approach.
+
+
+Compositing Multiple Layers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+It is also possible to combine data from various sources at the time of
+display.  Of course, one could use map algebra to produce a composite layer,
+but if the input layers are large, this could potentially be a time-consuming
+operation.  The TMS server allows for a list of sources to be supplied; these
+may be any combination of ``Pyramid`` objects and catalogs.  We then may
```

Comment:
Do you think this should link to `Pyrmaid` instead of just putting it in a block?